### PR TITLE
Add error monikers to the messageas generated by DNX/DNU

### PR DESCRIPTION
**Pull request by victorhurdugaci**
Fixes https://github.com/aspnet/dnx/issues/2440

Please review @davidfowl @lodejard @troydai @anurse

### CODE REVIEW

**Comment by anurse** [comment]
Can we centralize the error codes and message templates in one place and have a class with a bunch of static methods that create fully-formed Diagnostic Messages?

**Comment by moozzyk** [comment]
:+1: Other than that it looks good to me.

File: src/Microsoft.Dnx.Runtime.Internals/DiagnosticMonikers.cs [hunk 1]
```diff
@@ -0,0 +1,18 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.Dnx.Runtime.Internals
+{
+    internal class DiagnosticMonikers
+    {
+        public static readonly string NU1001 = nameof(NU1001); // The dependency A could not be resolved.
+        public static readonly string NU1002 = nameof(NU1002); // The dependency A in project B does not support framework C."
+        public static readonly string NU1003 = nameof(NU1003); // Invalid A section. The target B is invalid, targets must either be a file name or a directory suffixed with '/'. The root directory of the package can be specified by using a single '/' character.
```
Comment (troydai) [inline src/Microsoft.Dnx.Runtime.Internals/DiagnosticMonikers.cs]: Can we put the comments above the error code so as to shorten the line?

File: src/Microsoft.Dnx.Runtime.Abstractions/DiagnosticMessage.cs [hunk 2]
```diff
@@ -25,15 +26,17 @@ public DiagnosticMessage(string message, string filePath, DiagnosticMessageSever
...
+            string message,
+            string formattedMessage,
+            string filePath,
+            DiagnosticMessageSeverity severity,
+            int startLine,
+            int startColumn,
+            int endLine,
             int endColumn)
         {
+            ErrorCode = errorCode;
```
Comment (troydai) [inline src/Microsoft.Dnx.Runtime.Abstractions/DiagnosticMessage.cs]: Should we throw if the `errorCode` is `null`?

**Comment by moozzyk** [comment]
Actually are you planning to add any tests for this?

File: src/Microsoft.Dnx.Runtime.Abstractions/DiagnosticMessage.cs (hunk 2 above)
Comment (victorhurdugaci) [inline src/Microsoft.Dnx.Runtime.Abstractions/DiagnosticMessage.cs]: No. VS works well without it

**Comment by victorhurdugaci** [comment]
Will fix the test failures once we figure out what is the error format that VS understands

**Comment by victorhurdugaci** [comment]
@moozzyk  Testing this is pretty complicated if you want to go the DTH route. Maybe @troydai can help here.

I've updated the tests that cover error message so we get some coverage at least for the out of proc scenario.

**Comment by davidfowl** [comment]
@victorhurdugaci Did you figure out the parsing issue?

**Comment by victorhurdugaci** [comment]
@davidfowl check your e-mail :)

File: src/Microsoft.Dnx.Runtime.Abstractions/DiagnosticMessage.cs (hunk 2 above)
Comment (troydai) [inline src/Microsoft.Dnx.Runtime.Abstractions/DiagnosticMessage.cs]: All right.

**Comment by davidfowl** [comment]
:shipit: Once you test it with @BillHiebert 's new bits.

**Pull request merged**